File: test/c/EmbedIf.c
#include <stdio.h>
int main() {
    int x = 1;

    printf("Start\n");

    if (x == 1) {
        printf("Begin Embed\n");
        int y = 0;
        printf("End Embed\n");
    }

    printf("End\n");
}

File: test/java/EmbedIf.java
class EmbedIf
{
    public static void main(String args[])
    {
        int x = 1;
        System.out.println("Start");

        if (x == 1) {
            System.out.println("Begin Embed");
            int y = 0;
            System.out.println("End Embed");
        }

        System.out.println("End");
    }
}